### PR TITLE
Fix the missing intermediate CA certs on Citadel TLS service.

### PR DESCRIPTION
**Pull request by myidpt**
Fixes https://github.com/istio/istio/issues/16593.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by myidpt** [comment]
/hold

To add tests.

**Comment by JimmyCYJ** [comment]
We may need an integration test for this feature.

File: security/pkg/probe/controller_test.go [hunk 1]
```diff
@@ -34,8 +34,8 @@ import (
 
 func TestGcpGetServiceIdentity(t *testing.T) {
 	bundle, err := util.NewVerifiedKeyCertBundleFromFile(
-		"../pki/testdata/multilevelpki/int-cert.pem", "../pki/testdata/multilevelpki/int-key.pem",
-		"", "../pki/testdata/multilevelpki/root-cert.pem")
+		"../pki/testdata/multilevelpki/int2-cert.pem", "../pki/testdata/multilevelpki/int2-key.pem",
```
[the remark this answers is not in the document]
Comment (myidpt) [reply security/pkg/probe/controller_test.go]: Actually I don't need to update this test.

File: security/pkg/probe/controller.go [hunk 2]
```diff
@@ -98,7 +99,11 @@ func (c *LivenessCheckController) checkGrpcServer() error {
 	if signErr != nil {
 		return signErr.(ca.Error)
 	}
-
+	// Append the intermediate certs to the certificate to form a cert chain.
+	certChainPEM := c.ca.GetCAKeyCertBundle().GetCertChainPem()
+	if len(certChainPEM) > 0 {
+		certPEM = append(certPEM, certChainPEM...)
```
[the remark this answers is not in the document]
Comment (myidpt) [reply security/pkg/probe/controller.go]: Sign doesn't return the list. So I added SignWithCertChain.

File: security/pkg/server/ca/server.go [hunk 3]
```diff
@@ -262,7 +262,7 @@ func (s *Server) createTLSServerOption() grpc.ServerOption {
 		GetCertificate: func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
 			if s.certificate == nil || shouldRefresh(s.certificate) {
 				// Apply new certificate if there isn't one yet, or the one has become invalid.
```
Comment (incfly) [inline security/pkg/server/ca/server.go]: comment above needs updates as well.

**Comment by ameer00** [comment]
Confirming this works.  Tested with a non self-signed Citadel with mesh expansion betweek GKE and a GCE VM.

**Pull request merged**